TidepoolKitUI/LoginViewController.swift

```diff
@@ -0,0 +1,206 @@
+        passwordTextField.resignFirstResponder()
+        if (loginButton.isEnabled) {
+            loginButtonTapped(self)
+        }
+    }
+    
+    @IBAction func emailEnterHandler(_ sender: AnyObject) {
+        passwordTextField.becomeFirstResponder()
+    }
+    
+    @IBAction func cancelButtonTapped(_ sender: Any) {
+        tapOutsideFieldHandler(self)
+        if let loginSignupDelegate = loginSignupDelegate {
+            loginSignupDelegate.loginSignupCancelled()
+        } else {
+            self.dismiss(animated: true)
+        }
+    }
+    
+    @IBAction func loginButtonTapped(_ sender: AnyObject) {
+        updateButtonStates()
+        tapOutsideFieldHandler(self)
+        loginIndicator.startAnimating()
+        
+        //LogInfo("Logging into \(server?.rawValue ?? "default") server!")
+        tpKit.logIn(with: emailTextField.text!, password: passwordTextField.text!, serverHost: self.serverHost) {
+            result in
+            LogInfo("Login result: \(result)")
+            self.processLoginResult(result)
+        }
+    }
+    
+    private func processLoginResult(_ result: Result<TPSession, TidepoolKitError>) {
+        self.loginIndicator.stopAnimating()
+        switch result {
+        case .success(let session):
+            //LogInfo("Login success: \(user)")
+            self.logInComplete(session)
+        case .failure(let error):
+            var errorText = "Check your Internet connection!"
+            switch error {
+            case .unauthorized:
+                errorText = "Wrong email or password!"
+            default:
+                break
+            }
+            self.errorFeedbackLabel.text = errorText
+            self.errorFeedbackLabel.isHidden = false
+        }
+    }
+    
+    private func logInComplete(_ session: TPSession) {
+        if let loginSignupDelegate = loginSignupDelegate {
+            loginSignupDelegate.loginSignupComplete(session)
+        } else {
+            self.dismiss(animated: true)
+        }
+    }
+
+    @objc func textFieldDidChange() {
+        updateButtonStates()
+    }
+    
+    private func updateButtonStates() {
+        errorFeedbackLabel.isHidden = true
+        let connected = tpKit.isConnectedToNetwork()
+        // login button
+        if (emailTextField.text != "" && passwordTextField.text != "" && connected) {
+            loginButton.isEnabled = true
+            if #available(iOSApplicationExtension 13.0, *) {
+                loginButton.setTitleColor(UIColor.label, for:UIControl.State())
+            } else {
+                // Fallback on earlier versions
+                loginButton.setTitleColor(UIColor.black, for:UIControl.State())
+            }
+        } else {
+            loginButton.isEnabled = false
+            loginButton.setTitleColor(UIColor.gray, for:UIControl.State())
+        }
+    }
+    
+    private var currentServerHost: String {
+        return self.serverHost ?? tpKit.currentServerHost
```

Comment:
It already passes back the selected server host in the session object. The problem is that there are two "default" server hosts: the last selected one (for debug login) and the production host, which is not returned by the dynamic DNS SRV record lookup. 

So I changed TidepoolKit to be ignorant of any of these, and to just require a server host to be passed into login. I also changed TidepoolKitUI to ensure that the production server host appears in the debug UI host selection list (test scenario where a non-production host has been selected previously, but now the user wants to go back to production, but the DNS lookup does not provide this). The client of TidepoolKitUI will now also have to conjure up a server host if it has not yet persisted one: this should be production for shipping code. 

I'm not overly fond of this solution because now the production server host needs to be hardwired in each client of TidepoolKitUI as well as in TidepoolKitUI itself (before this change it was just kept in TidepoolKit) - so, open to suggestions here...